osidb/sync_manager.py

```diff
@@ -328,11 +365,58 @@ def check_for_reschedules(cls):
                 )
                 continue
 
+    @classmethod
+    def is_in_progress(cls, sync_id):
+        """
+        Check if there is a task running for this sync manager. Task is considered running
+        if it has started and not finished yet or failed.
+        """
+
+        manager = cls.objects.get(sync_id=sync_id)
+
+        # This function is called before "last_started_dt" is set and
+        # a failed task has does not have "last_finished_dt" set
+        #
+        # 1) If "last_started_dt" is None, we assume that the task is NOT running.
+        # 2) If "last_started_dt" is not None, "last_failed_dt" is None, and "last_finished_dt" is None,
+        #    the task has started and is currently IN PROGRESS
+        # 3) If "last_started_dt" is not None, "last_failed_dt" is not None (and > last_started_dt),
+        #    and "last_finished_dt" is None, the previous run FAILED
+        # 4) If "last_started_dt" is not None and "last_finished_dt" is not None (and > last_started_dt),
+        #    the task has COMPLETED successfully
+
+        return (
+            # must have started
+            manager.last_started_dt is not None
+            and
+            # must NOT have failed since that start
+            not (
+                manager.last_failed_dt is not None
+                and manager.last_failed_dt > manager.last_started_dt
+            )
+            and
+            # and either not finished (-> running) or explicitly in-progress
+            manager.last_finished_dt is None
+        )
```

Comment:
This is an estimation in the sense that we assume that when started a task must either finish or fail with respect to the sync manager framework but if Celery crashes etc. there will be no finishing or failing. This is probably as much as we can get without going on the Celery level though.